ddl/ddl_api.go

```diff
@@ -3554,6 +3563,12 @@ func buildPartitionInfo(meta *model.TableInfo, d *ddl, spec *ast.AlterTableSpec)
 		}
 		// For RANGE partition only VALUES LESS THAN should be possible.
 		clause := def.Clause.(*ast.PartitionDefinitionClauseLessThan)
+		if len(part.Columns) > 0 {
+			if err := checkRangeColumnsTypeAndValuesMatch(ctx, meta, part.Columns, clause.Exprs); err != nil {
```

Comment:
Could we put it to `checkPartitionByRange`? Then we can remove the `if s != nil ` in line 1621

Reply:
Partition by range do not check value and column type match like partition by range columns.

Reply:
Partition by range checks value expression evaluates to an integer...

Reply:
Got it.